src/components/GiftRankingBox.tsx

```diff
@@ -30,31 +30,21 @@ type GiftRankingProps = {
 
 const GiftRanking = ({ target, rankType }: GiftRankingProps) => {
   const theme = useTheme();
-  const [productRankingData, setProductRankingData] = useState<
-    ProductRanking[] | null
-  >(null);
-  const [isLoading, setIsLoading] = useState(true);
-  const [isError, setIsError] = useState(false);
 
-  const sortParam = `/products/ranking?targetType=${target}&rankType=${rankType}`;
-
-  useEffect(() => {
-    const fetchProductRanking = async () => {
-      try {
-        const response = await api.get(sortParam);
-        setProductRankingData(response.data.data);
-      } catch (error) {
-        console.error("Error fetching theme data:", error);
-        setIsError(true);
-      } finally {
-        setIsLoading(false);
-      }
-    };
+  const productsFiltered = `/products/ranking?targetType=${target}&rankType=${rankType}`;
+  const navigate = useNavigate();
 
-    fetchProductRanking();
-  }, [sortParam]);
+  const {
+    data: productRankingData,
+    isError,
+    isLoading,
+  } = useQuery({
+    queryKey: ["productRanking", target, rankType],
+    queryFn: () => api.get(productsFiltered),
```

Comment:
```suggestion
  type BaseResponse<T> = {
   data: T
   }
    queryFn: () => api.get<BaseResponse<ProductRanking[]>>(productsFiltered),
```

이렇게 타입을 주입해주시면 타입이 자연스럽게 추론되어서, 밑에서 타입을 추가해주시지 않으셔도 될거 같아요!